server/main_server/src/main.cpp

```diff
@@ -0,0 +1,1342 @@
+    auto launch_delay_server(bool _write_to_stdout, bool _enable_test_mode) -> void;
+    auto migrate_and_launch_delay_server(bool _write_to_stdout,
+                                         bool _enable_test_mode,
+                                         std::chrono::steady_clock::time_point& _time_start) -> void;
+    auto log_stacktrace_files(std::chrono::steady_clock::time_point& _time_start) -> void;
+    auto remove_leftover_agent_info_files_for_ips() -> void;
+
+    auto is_server_listening_for_connections(const std::string& _host, const std::string& _port) -> int;
+} // anonymous namespace
+
+auto main(int _argc, char* _argv[]) -> int
+{
+    [[maybe_unused]] const auto boot_time = std::chrono::system_clock::now();
+
+    bool write_to_stdout = false;
+    bool enable_test_mode = false;
+
+    ProcessType = SERVER_PT; // This process identifies itself as a server.
+
+    set_ips_display_name("irodsServer");
+
+    namespace po = boost::program_options;
+
+    po::options_description opts_desc{""};
+
+    // clang-format off
+    opts_desc.add_options()
+        ("daemonize,d", "")
+        ("pid-file,p", po::value<std::string>(), "")
+        ("stdout", po::bool_switch(&write_to_stdout), "")
+        ("test-mode,t", po::bool_switch(&enable_test_mode), "")
+        ("help,h", "")
+        ("version,v", "");
+    // clang-format on
+
+    try {
+        po::variables_map vm;
+        po::store(po::command_line_parser(_argc, _argv).options(opts_desc).run(), vm);
+        po::notify(vm);
+
+        if (vm.count("help") > 0) {
+            print_usage();
+            return 0;
+        }
+
+        if (vm.count("version") > 0) {
+            print_version_info();
+            return 0;
+        }
+
+        if (vm.count("daemonize") > 0) {
+            daemonize();
+        }
+
+        std::string pid_file = (irods::get_irods_runstate_directory() / "irods/irods-server.pid").string();
+        if (const auto iter = vm.find("pid-file"); std::end(vm) != iter) {
+            pid_file = std::move(iter->second.as<std::string>());
+        }
+
+        if (create_pid_file(pid_file) != 0) {
+            fmt::print(stderr, "Error: could not create PID file [{}].\n", pid_file);
+            return 1;
+        }
+    }
+    catch (const std::exception& e) {
+        fmt::print(stderr, "Error: {}\n", e.what());
+        return 1;
+    }
+
+    try {
+        if (!validate_configuration()) {
+            return 1;
+        }
+
+        const auto config_file_path = irods::get_irods_config_directory() / "server_config.json";
+        irods::server_properties::instance().init(config_file_path.c_str());
+        irods::environment_properties::instance(); // Load the local environment file.
+
+        // TODO Consider removing the need for these along with all options.
+        // All logging should be controlled via the new logging system.
+        rodsLogLevel(LOG_NOTICE);
+    	rodsLogSqlReq(0);
```

Comment:
Close this comment once the issue is opened. See the comment for the same thing in agent_main.cpp.

Reply:
Created #8002

Reply:
Done.